src/handlers/TechRecordsListHandler.ts

```diff
@@ -0,0 +1,165 @@
+import { Vehicle } from "../../@Types/TechRecords";
+import { ErrorHandler } from "./ErrorHandler";
+
+export class TechRecordsListHandler<T extends Vehicle> {
+  private readonly techRecordsDAO: TechRecordsDAO;
+
+  constructor(techRecordsDAO: TechRecordsDAO) {
+    this.techRecordsDAO = techRecordsDAO;
+  }
+
+  /* #region  Public functions */
+  public async getFormattedTechRecordsList(
+    searchTerm: string,
+    status: string,
+    searchCriteria: ISearchCriteria = SEARCHCRITERIA.ALL
+  ): Promise<T[]> {
+    try {
+      // Formatting the object for lambda function
+      let techRecordItems = await this.getTechRecordList(
+        searchTerm,
+        status,
+        searchCriteria
+      );
+      techRecordItems = this.formatTechRecordItemsForResponse(techRecordItems);
+
+      return techRecordItems;
+    } catch (error) {
+      if (!(error instanceof HTTPError)) {
+        console.error(error);
+        error.statusCode = 500;
+        error.body = HTTPRESPONSE.INTERNAL_SERVER_ERROR;
+      }
+      throw ErrorHandler.Error(error.statusCode, error.body);
+    }
+  }
+
+  public async getTechRecordList(
+    searchTerm: string,
+    status: string,
+    searchCriteria: ISearchCriteria = SEARCHCRITERIA.ALL
+  ): Promise<T[]> {
+    const data: any = await this.techRecordsDAO.getBySearchTerm(
+      searchTerm,
+      searchCriteria
+    );
+    if (data.Count === 0) {
+      throw new HTTPError(404, HTTPRESPONSE.RESOURCE_NOT_FOUND);
+    }
+    // Formatting the object for lambda function
+    let techRecordItems: T[] = data.Items;
+    if (status !== STATUS.ALL) {
+      techRecordItems = this.filterTechRecordsByStatus(techRecordItems, status);
+    }
+    return techRecordItems;
+  }
+
+  public formatTechRecordItemForResponse(techRecordItem: T) {
+    // Adding primary and secondary VRMs in the same array
+    const vrms = [];
+    if (techRecordItem.primaryVrm) {
+      vrms.push({ vrm: techRecordItem.primaryVrm, isPrimary: true });
+    }
+    if (techRecordItem.secondaryVrms) {
+      for (const secondaryVrm of techRecordItem.secondaryVrms) {
+        vrms.push({ vrm: secondaryVrm, isPrimary: false });
+      }
+    }
+    Object.assign(techRecordItem, {
+      vrms
+    });
+    // Cleaning up unneeded properties
+    delete techRecordItem.primaryVrm; // No longer needed
+    delete techRecordItem.secondaryVrms; // No longer needed
+    delete techRecordItem.partialVin; // No longer needed
+
+    techRecordItem.techRecord.forEach((techRecord: any) => {
+      if (
+        techRecord.euroStandard !== undefined &&
+        techRecord.euroStandard !== null
+      ) {
+        techRecord.euroStandard = techRecord.euroStandard.toString();
+      }
+    });
+
+    return techRecordItem;
+  }
+  /* #endregion */
+
+  /* #region  Private functions */
+  private filterTechRecordsByStatus(techRecordItems: T[], status: string): T[] {
+    const recordsToReturn = [];
+    for (let techRecordItem of techRecordItems) {
```

Comment:
what about
```
// we could be looking at extending it later and passing status a separate argument if we wanted to perform separate operations depending on the status

techRecordItems
  .map(techRecord => this.filterTechRecordsForIndividualVehicleByStatus(techRecord, status))
```
would this work?

Also have a question does it matter whether we have at least 1 record as we would throw an error if we can't find/have no tech-record  or would return undefined I believe ?